packages/lib/src/vis/surface/hooks.ts

```diff
@@ -2,31 +2,28 @@ import type { NumArray } from '@h5web/shared';
 import { getDims } from '@h5web/shared';
 import type { NdArray } from 'ndarray';
 import { useMemo } from 'react';
-import { Float32BufferAttribute, Uint8BufferAttribute } from 'three';
+import { Float32BufferAttribute } from 'three';
 
-export function useBufferAttributes(
-  dataArray: NdArray<NumArray>,
-  dataToColorScale: (val: number) => [number, number, number]
-) {
-  const { position, color } = useMemo(() => {
+export function useBufferAttributes(dataArray: NdArray<NumArray>) {
+  const { position, uv } = useMemo(() => {
     const { rows, cols } = getDims(dataArray);
 
     const positions: number[] = [];
-    const colors: number[] = [];
+    const uvs: number[] = [];
 
     for (let row = 0; row < rows; row++) {
       for (let col = 0; col < cols; col++) {
         const value = dataArray.get(row, col);
         positions.push(col, row, value);
-        colors.push(...dataToColorScale(value));
+        uvs.push((col + 0.5) / cols, (row + 0.5) / rows);
```

Comment:
Not really understood why but there was a shift of `[0.5, 0.5]` between the applied color and the point it referred to :shrug:

Reply:
After investigation, I think this is because in `HeatmapVis`, the "pixel" color is applied to the center that is between the data points while we want the color to be applied to the data point for the `SurfaceVis`.

If you checkout the branch, you can check this yourself by setting `showPoints` to `true` and removing the `filter` of `HeatmapMaterial` in `SurfaceMesh`.